File: app/src/main/java/com/personal/lifeOS/features/profile/domain/model/UserProfile.kt
package com.personal.lifeOS.features.profile.domain.model

data class UserProfile(
    val name: String = "",
    val email: String = "",
    val phone: String = "",
    val avatarInitials: String = "?",
    val profilePicUri: String = "",
    val memberSince: Long = 0L, // epoch millis of first profile creation
    val isBiometricEnabled: Boolean = false,
    val notificationsEnabled: Boolean = true,
)

File: lib/features/profile/domain/entities/user_profile.dart
class UserProfile {
  const UserProfile({
    required this.name,
    required this.email,
    required this.phone,
    required this.memberSinceLabel,
    required this.verified,
    this.avatarUrl,
  });

  final String name;
  final String email;
  final String phone;
  final String memberSinceLabel;
  final bool verified;
  final String? avatarUrl;
}
